### PR TITLE
fix: Bump phx to 1.7.7 to potentially reduce memory usage

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?

https://github.com/phoenixframework/phoenix/issues/4933

We have some cowboy connections that seem to not be sweeped after a certain amount of time, this could be due to the issue found above which has a fix on phx 1.7.7

Other code changes are for warning removal.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1,3 @@
+elixir 1.14.3-otp-25
+nodejs 18.13.0
+erlang 25.3
```
Comment (abc3) [inline .tool-versions:3]: why do we need this file?

Comment (filipecabaco) [reply .tool-versions:3]: to use asdf to install the required versions to run realtime. makes it easier to onboard people to the project if need be.

with this and asdf we just need to do `asdf install` to have the dev environment setup

Comment (abc3) [reply .tool-versions:3]: sounds good 👍

**Review by abc3 — APPROVED** [review]

**Pull request merged**